internal/server/email/sendgrid.go

```diff
@@ -0,0 +1,84 @@
+package email
+
+import (
+	"errors"
+	"os"
+
+	"github.com/sendgrid/sendgrid-go"
+	"github.com/sendgrid/sendgrid-go/helpers/mail"
+
+	"github.com/infrahq/infra/internal/logging"
+)
+
+type EmailTemplate int8
+
+const (
+	EmailTemplateAccountCreated EmailTemplate = iota
+	EmailTemplatePasswordReset
+	EmailTemplateUserInvite
+)
+
+var emailTemplateIDs = map[EmailTemplate]string{
+	EmailTemplateAccountCreated: "",
+	EmailTemplatePasswordReset:  "d-d87873d2f11b4055befbd7064cda44d6", // transactional-password-reset
+	EmailTemplateUserInvite:     "d-ac6bcb2a7f02463c8b7fc8caffc35f2d", // transactional-user-invite // TODO: Should this instead be a "granted access to x" email?
+}
+
+var (
+	AppDomain          = "https://infrahq.com"
+	FromAddress        = "noreply@infrahq.com"
```

Comment:
I guess both of these will need to be part of `server.Options` to support on-prem.